cpp/proto/xjson_serializer.cc

```diff
@@ -41,7 +41,12 @@ SerializeResult CheckJsonFormat(const string& json) {
 
 string V1LeafData(const LogEntry& entry) {
   CHECK(entry.has_x_json_entry());
-  return entry.x_json_entry().json();
+  hash h;
+  if (!common_json_hash(entry.x_json_entry().json().c_str(), h)) {
+	LOG(ERROR) << "Faild to compute object hash: "
+	           << entry.x_json_entry().json() << ".";
+  }
+  return string(reinterpret_cast<const char*>(&h[0]), sizeof(h));
```

Comment:
That's a fairly different implementation than the other ones, no? So the leaf hash preimage for XJSON is... the hash of the object? I guess objecthash doesn't give you access to its normalised representation, so you have to go with what you've got?